c2rust-ast-exporter/src/AstExporter.cpp

```diff
@@ -323,7 +323,12 @@ class TypeEncoder final : public TypeVisitor<TypeEncoder> {
 
 #if CLANG_VERSION_MAJOR >= 10
         // Handle built-in vector types as if they're normal vector types
-        if (kind >= BuiltinType::SveInt8 && kind <= BuiltinType::SveBool) {
+        if (kind >= BuiltinType::SveInt8 && kind <= BuiltinType::SveBool
+#if CLANG_VERSION_MAJOR >= 13
+            /* RISC-V vector types */
+            || kind >= BuiltinType::RvvInt8mf8 && kind <= BuiltinType::RvvBool64
+#endif // CLANG_VERSION_MAJOR >= 13
+            ) {
```

Comment:
```suggestion
        const bool is_sve = kind >= BuiltinType::SveInt8 && kind <= BuiltinType::SveBool;
#else
        const bool is_sve = false;
#endif // CLANG_VERSION_MAJOR >= 10

#if CLANG_VERSION_MAJOR >= 13
        // RISC-V vector types
        const bool is_rvv = kind >= BuiltinType::RvvInt8mf8 && kind <= BuiltinType::RvvBool64;
#else
        const bool is_rvv = false;
#endif // CLANG_VERSION_MAJOR >= 13

#if CLANG_VERSION_MAJOR >= 10
            if (is_sve || is_rvv) {
```

I think this is cleaner and more symmetrical.

I don't *think* that last `#if CLANG_VERSION_MAJOR >= 10` is needed, so if it's not, you can remove it and its `#endif`.

This change I think should go in a separate commit.

Reply:
The "if >= 10" is still needed for switch-cases that refer to `builtinType::Sve*`. I don't think of this as strictly better, but it's not worse either, so as you prefer it, I'm working it in.

Reply:
The diff algorithm made 3001e7b77b957903bef237fc6e44d8ff7880d693 remarkably pretty :-)

Reply:
(Not *that* slim any more as 0feb8f3b091711a2897c1261eeaee73e1007ac6d now that I adjusted the comments, but still looks fine.)

Reply:
This looks much better I think!  Narrowing the scope of the `#ifdef`s helps a lot in understanding the code.  And the comments at the top are rearranged better now, too.

Reply:
This has been causing all kinds of other trouble, compared to the original proposal.

I'm splitting that out into another PR (https://github.com/immunant/c2rust/pull/694) for the record, but think this one is better off without them. It is rolled back to only the original changes, with all the smaller fixes, but without this thread's refactoring.

Reply:
Thanks, I agree your way is better.